backend/app/api/v1/profile.py

```
@@ -0,0 +1,110 @@
+    print(f"Extracted token: {token}")
+    result = await validate_user(token)
+    if not result:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid or expired token"+token,
+        )
+    return result  
+
+
+@router.get("/")
+async def fetch_details(user=Depends(get_user),):
```

Ruff (0.14.11)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)